apps/studio/prisma/schema.prisma

```diff
@@ -26,68 +26,77 @@ model VerificationToken {
-
-  // To return `Date`s intact through the API we use transformers
-  // https://trpc.io/docs/data-transformers
-  // This is unique so it can be used for cursor-based pagination
-  createdAt DateTime  @default(now())
-  updatedAt DateTime  @default(now()) @updatedAt
-  deletedAt DateTime? @db.Timestamp
-
-  @@index([parentPostId])
-  @@index([parentPostId, deletedAt, id])
-  @@index([deletedAt])
-  @@index([createdAt])
+model Resource {
+  id             String       @id @default(cuid())
+  name           String
+  siteId         String
+  parentId       String?
+  lastEditTime   DateTime?
+  lastEditUserId String
+  blobId         String?      @unique
+  variant        ResourceType
+  status         PageStatus // Is this a good place to be storing this?
+  site           Site         @relation(fields: [siteId], references: [id])
+  parent         Resource?    @relation("parentRelation", fields: [parentId], references: [id])
+  children       Resource[]   @relation("parentRelation")
+  lastEditUser   User?        @relation(fields: [lastEditUserId], references: [id], onDelete: NoAction, onUpdate: NoAction) // Don't delete the file if the user is deleted
+  blob           Blob?
+  permission     Permission[]
+
+  @@unique([siteId, parentId, name], name: "uniqueChildNames")
 }
 
-model LikedPosts {
-  post      Post     @relation(fields: [postId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  postId    String
-  user      User     @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  userId    String
-  createdAt DateTime @default(now())
+model User {
+  id    String @id @default(cuid())
+  name  String
+  email String @unique
+  phone String
 
-  @@id([postId, userId])
+  Resource   Resource[]
+  Permission Permission[]
 }
 
-model User {
-  id            String       @id @default(cuid())
-  name          String?
-  username      String?      @unique
-  email         String?      @unique
-  emailVerified DateTime?
-  image         String?
-  bio           String?
-  posts         Post[]
-  likedPosts    LikedPosts[]
-  accounts      Accounts[]
+model Permission {
+  id         String   @id @default(cuid())
+  resourceId String
+  userId     String
+  role       RoleType
+
+  // Delete the row if either resource or user is deleted, default behavior cascade
+  resource Resource @relation(fields: [resourceId], references: [id])
+  user     User     @relation(fields: [userId], references: [id])
+
+  // userId and resource Id should be unique? Each user should have unambiguous access to resource? Or not if we take the strictest or laxest perm for the user
+  @@unique([userId, resourceId], name: "unambiguousPermissionsPerUserResource")
 }
 
-model Accounts {
-  id                String @id @default(cuid())
-  provider          String
-  providerAccountId String
+model Site {
+  id   String @id @default(cuid())
+  name String @unique
+
+  Resource Resource[]
```

Comment:
should `Site` have a 1 to many key to user indicating that the user has access to this site?

Reply:
hmmm, from your experience, is it more natural to store this relation in the Site table or the User table?

Reply:
Both are ok, we could even have a join table.

Reply:
many to many?